### DISCUSSION TITLE
Time section is not automatically refreshing

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [solidtime-io/solidtime#246] [Feature requests]
Sometimes I happen to run off from my PC to do something only to remember that I have left timer going. No problem, I just need to log in from my phone and end the timer - that works fine. The issue is that when I go back to my PC, the timer is stopped, but I do not see the last stopped record. The list/table is not refreshing automatically and I need to manually refresh the whole page.

It would be great to have real-time sync there.

**Comment by korridor** [comment]
We decided to move feature requests into github discussions, to separate feature requests from bugs and allow for better prioritization. If this feature is important to you, you can upvote it to let us know.

**Reply by FluffyDiscord** [reply]
This shouldn't be a "feature request", or are we using the "it's not a bug, it's a feature"?

**Reply by Onatcer** [reply]
you really need to tone down your comments here, this is an open source project, if you are not happy with how we handle things here you can fork it and develop your own one. I clarified below under what circumstances i would consider it a bug. I would kindly ask you to engage in a respectful conversation with the people who are actually maintaining and working on this project.

the fact that you used "real-time sync" made it seem like to me that you are talking about having the page open, while using the app from another device. This case is not supported and would require either automatic revalidation or websockets, which are a feature. We generally refresh the state when you focus the page again which solves most of the state problems, but it is not a real-time sync. 

I checked it now; you are correct it is refreshing the way it is intended due to the infinite scroll on the page. I will look into a solution.

**Reply by FluffyDiscord** [reply]
Since the timer was stopped (somehow that synced just fine), but the ended/saved record was not in the table bellow (this apparently did not sync), I was under the impression that it is a bug, because "half" of the UI/state did actually refresh and the other "half" not.

That was the reason for my sharp tongue, since if we consider the behavior that I described, one can assume that it's a bug. I still do think that it's a bug, because if you can refresh the timer state (on/off), you should be able to refresh the table records too using the same event/logic.

If the records table, and the timer are somehow separate things that cannot be refreshed together as you describe on page focus, and it would need websockets or pushevents then yes, that would be a feature.

Reading the words of your colleague I felt that he just wanted to push the issue away, hoping it would eventually be forgotten.

Also little note on the "open source" - maybe allow community members to contribute?

**Comment by Onatcer** [comment]
data should generally refresh when you focus on the window again; if it doesn't do that in some scenarios, I would consider that a bug, but I would need a reproducible example.

**Comment by FluffyDiscord** [comment]
The timer does refresh and stops, but not the records in the table.

**Comment by Onatcer** [comment]
For reference this should address the problem, it is not deployed nor tagged yet. 

https://github.com/solidtime-io/solidtime/commit/df0f3b26806c4def6bd3f6717f7450c5efdb7c51
